### DISCUSSION TITLE
404 NOT FOUND - at the second attempt of removal, the second post.

### DISCUSSION THREAD

**Discussion by xp0nat** [livewire/livewire#6390] [1. Help] [answered]
When I try to delete posts, when I delete any - everything is fine. As soon as I try to delete another post right after that, I get: 404
NOT FOUND. Before upgrading to version 3 everything was fine. I've been looking into upgrading to version 3, but there was some confusion with that. Hope someone can see the solution, thanks!

    public function delete($postId)
    {
        $post = Post::findOrFail($postId);
        $post->delete();
    }
    
    <button  wire:click.prevent="delete({{ $post->id }})">{{ trans('post.delete') }}</button>

**Comment by danie-ramdhani** [comment]
maybe the post doesn't exist? because `Post::findOrFail($postId)` is throwing 404 if fail.

also you can try dependency injection:

```php
public function delete(Post $post)
{
    $post->delete();
}
```

```blade
<button wire:click="delete({{ $post->id }})">{{ trans('post.delete') }}</button>
```

**Reply by xp0nat** [reply]
Thank you for your answer.

Unfortunately the same error. Let's say I have 3 posts. When I remove the first one, I can also remove the third one, but I can't remove the second one because I get a 404
NOT FOUND. It's really weird, no idea.

Also, after reloading the page of course I can remove it. I tried to force a component refresh after removal, but that didn't work for me either.

**Reply by danie-ramdhani** [reply, marked as answer]
maybe add wire key in the button:

```blade
<button wire:click="delete({{ $post->id }})" wire:key="{{ $post->id }}">{{ trans('post.delete') }}</button>
```

**Reply by xp0nat** [reply]
When I used $post instead of $post->id, everything is the same, I can delete the first and third, but the second gives me a new error:
"Array to string conversion"

<button wire:click="delete({{ $post }})">{{ trans('post.delete') }}</button>

**Reply by xp0nat** [reply]
This really worked for me! Thank you very much! I don't know why in version 3 liveware doesn't work as it used to without a key, but with the addition of wire:key="{{ $post->id }} everything is ok. Thanks again.

**Reply by danie-ramdhani** [reply]
because livewire now is not using morphdom like v2. see the references in here: https://livewire.laravel.com/docs/morphing

**Reply by xp0nat** [reply]
this really needs more of my attention with an upgrade, Thank you.

**Reply by xp0nat** [reply]
it also works if you use it in that place, without being reused in different scenarios.
 
 @forelse ($posts as $post)
< div wire:key="{{ $post->id }}" >

**Comment by xp0nat** [comment]
> maybe the post doesn't exist? because `Post::findOrFail($postId)` is throwing 404 if fail.
> 
> also you can try dependency injection:
> 
> ```
> public function delete(Post $post)
> {
>     $post->delete();
> }
> ```
> 
> ```
> <button wire:click="delete({{ $post->id }})">{{ trans('post.delete') }}</button>
> ```
Thank you for your answer.

Unfortunately the same error. Let's say I have 3 posts. When I remove the first one, I can also remove the third one, but I can't remove the second one because I get a 404
NOT FOUND. It's really weird, no idea.


> maybe the post doesn't exist? because `Post::findOrFail($postId)` is throwing 404 if fail.
> 
> also you can try dependency injection:
> 
> ```
> public function delete(Post $post)
> {
>     $post->delete();
> }
> ```
> 
> ```
> <button wire:click="delete({{ $post->id }})">{{ trans('post.delete') }}</button>
> ```



> ```
> <button wire:click="delete({{ $post->id }})" wire:key="{{ $post->id }}">{{ trans('post.delete') }}</button>
> ```
